File: JavaAES/src/test/java/com/dmeritano/javaaes/AESChiperTest.java
package com.dmeritano.javaaes;

import static org.junit.Assert.assertEquals;
import org.junit.Before;
import org.junit.Test;

public class AESChiperTest {
    
    private String key;
    private String textToEncrypt;
    private String textEncrypted;
    
    @Before
    public void setUp() throws Exception {
        key = TestUtilities.getAlphaNumericRandomString(10);
        textToEncrypt = TestUtilities.getAlphaNumericRandomString(50);
        textEncrypted = AESChiper.encrypt(textToEncrypt, key);
    }
    
     @Test
    public void testEncrypt() throws Exception {
        assertEquals(textEncrypted, AESChiper.encrypt(textToEncrypt, key));
    }

    @Test
    public void testDecrypt() throws Exception {
        assertEquals(textToEncrypt, AESChiper.decrypt(textEncrypted, key));
    }   
    
}

File: CSharpAES/CSharAESTest/AESChiperTest.cs
using System;
using CSharpAES;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CSharAESTest
{
    [TestClass]
    public class AESChiperTest
    {
        private string key;
        private string data;
        private string dataEncrypted;

        public AESChiperTest()
        {
            key = TestsUtilities.GetAlphaNumericRandomString(10);
            data = TestsUtilities.GetAlphaNumericRandomString(50);
            dataEncrypted = AESChiper.Encrypt(data, key);
        }

        [TestMethod]
        public void testEncrypt()
        { 
            Assert.AreEqual(dataEncrypted, AESChiper.Encrypt(data, key));
        }

        [TestMethod]
        public void testDecrypt()
        {
            Assert.AreEqual(data, AESChiper.Decrypt(dataEncrypted, key));
        }
    }
}
